src/realm/alloc_slab.cpp

```diff
@@ -889,9 +894,12 @@ ref_type SlabAlloc::attach_file(const std::string& path, Config& cfg, util::Writ
 
     reset_free_space_tracking();
     update_reader_view(size);
-    REALM_ASSERT(m_mappings.size());
-    m_data = m_mappings[0].primary_mapping.get_addr();
-    realm::util::encryption_read_barrier(m_mappings[0].primary_mapping, 0, sizeof(Header));
+    {
```

Comment:
Locking here should not be needed. The slab allocator is just being attached, and no other thread should have a reference to it at this point.